app/models/activity.rb

```diff
@@ -18,9 +18,60 @@ def event_key_label
     event_key.humanize.upcase
   end
 
-  def response_code
-    return "no message" unless message
-    # return  "no response" unless message["response_metadata"]
-    JSON.parse(message.to_json).keys
+  def application_payload
+    return {} unless message
+    payload = JSON.parse(message.to_json, symbolize_names: true)
+    payload[:application]
   end
+
+  def request?
```

Comment:
I think this method should return a boolean value if we're sticking with the Ruby convention for method names ending in "?".  Beyond just convention, though, it could be confusing since this method would return true for an empty string.